src/snowflake/snowpark/session.py

```diff
@@ -637,6 +639,16 @@ def _analyzer(self) -> Analyzer:
             )
         return self._thread_store.analyzer
 
+    @property
+    def _plan_builder(self):
+        if not hasattr(self._thread_store, "plan_builder"):
```

Comment:
oh, i didn't notice we have plan builder here, i thought plan builder is just part of the analyzer, what is the plan builder used here?, and what is the reason that we moved it to a property

Reply:
and the plan builder seems thread local to me

Reply:
This plan builder is different from analyzer's plan builder. Only used by
1. DataFramReader to create read_file plan
2. FileOperation to create file_operation_plan

Reply:
I added it as part of old changes. reverted back to original now since plan builder should be thread-safe with restricted config updates